### DISCUSSION TITLE
Change the text color of the days in the previous and following months inside the RangePicker component

### DISCUSSION THREAD

**Discussion by Zexol11** [ant-design/ant-design#44773] [Q&A]
I want to change the following Saturday and Sunday text colors on the RangePicker, but my condition is in the Start Dates and End Dates of the RangePicker, the Saturday and Sunday days from the last month and following month that is in that DatePicker calender should have a lighter color. How can I achieve this?

[Stackblitz Code Sample, not finished](https://stackblitz.com/edit/react-jtpazq?file=demo.js)

[Sample image 1](https://i.stack.imgur.com/2Pvuo.png)
[Sample image 2](https://i.stack.imgur.com/eB0zx.png)

For example:

Start Date Month (June):

June's Saturday and Sunday = Blue and red
July's Saturday and Sunday = light blue and light red
May's Saturday and Sunday = light blue and light red
End Date Month (July):

July's Saturday and Sunday = Blue and red
August's Saturday and Sunday = light blue and light red
June's Saturday and Sunday = light blue and light red
Thank you in advance!

**Comment by Wxh16144** [comment]
It's too complicated, I'll try to give a copy of my solution, not sure if it meets your expectations.

```tsx
import { DatePicker } from 'antd';
import type { RangePickerProps } from 'antd/es/date-picker';
import type { Dayjs } from 'dayjs';
import dayjs from 'dayjs';
import * as React from 'react';
import useMergeState from 'use-merge-value';

const MyRangePicker = (props: RangePickerProps, ref: any) => {
  const [value, setValue] = useMergeState<[Dayjs, Dayjs]>(
    [dayjs(), dayjs()],
    props as any,
  );

  const cellRender: RangePickerProps['cellRender'] = (current, info) => {
    if (props.cellRender) return props.cellRender(current, info);
    if (info.type !== 'date') return info.originNode;

    const [start, end] = value ?? [];

    const style: React.CSSProperties = {};
    if (current.day() === 0) {
      if (current.isBefore(start, 'month') || current.isAfter(end, 'month')) {
        style.color = 'lightblue';
      } else {
        style.color = 'blue';
      }
    } else if (current.day() === 6) {
      if (current.isBefore(start, 'month') || current.isAfter(end, 'month')) {
        style.color = 'lightgreen';
      } else {
        style.color = 'green';
      }
    }

    return React.cloneElement(info.originNode, { style });
  };

  return (
    <DatePicker.RangePicker
      ref={ref}
      {...props}
      value={value}
      onChange={setValue}
      cellRender={cellRender}
    />
  );
};

export default React.forwardRef(MyRangePicker);

```

**Reply by Zexol11** [reply]
Thank you for your response, it seems that I'll abandon this implementation, As you've said, it is indeed kind of complicated.
